### PR TITLE
[sc-4395] add values override flag to deploy command

### PR DESCRIPTION
**Pull request by steb-veksler**

### CODE REVIEW

**Review by noamApps — COMMENTED** [review]
will the user provided values will override everything? (including our genenrated resources values) I think they should

File: pkg/helm/values.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+		if err = yaml.Unmarshal(data, chartValues); err != nil {
+			return nil, err
+		}
+	}
+
+	return valuesOverride, nil
+}
+
+func readValuesOverride(path string) ([]byte, error) {
```
Comment (noamApps) [inline pkg/helm/values.go:36]: is this really needed ? like a preparation for fetching presets from our docs or something?

File: pkg/helm/values.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	var err error
+
+	valuesOverride := make(map[string]interface{})
+
+	for _, path := range paths {
+		var data []byte
+		if data, err = readValuesOverride(path); err != nil {
+			return nil, err
+		}
+
```
Comment (noamApps) [inline pkg/helm/values.go:23]: what is the meaning of this double unmarshaling?

**Review by noamApps — COMMENTED** [review]
didn't completely understand if this guarantees that the user provided values will be the most significant ?

File: pkg/helm/values.go (hunk 2 above)
Comment (steb-veksler) [reply pkg/helm/values.go:23]: needed to get only the user override values for sentry context

**Comment by steb-veksler** [comment]
> didn't completely understand if this guarantees that the user provided values will be the most significant ?

the execution order of yaml Uunmarshal  guarantees that the user provided values will be the most significant

default values -> resources presets -> user overrides

`		chartValues := defaultChartValues(clusterName, apiKey.ApiKey)
		if sentryHelmContext.ResourcesPresets, err = helm.TuneResourcesValues(&chartValues, adequateNodesReports); err != nil {
			return err
		}
		if sentryHelmContext.ValuesOverride, err = helm.LoadChartValuesOverrides(&chartValues, viper.GetStringSlice(VALUES_FLAG)); err != nil {
			return err
		}`

**Comment by noamApps** [comment]
> > didn't completely understand if this guarantees that the user provided values will be the most significant ?
> 
> the execution order of yaml Uunmarshal guarantees that the user provided values will be the most significant
> 
> default values -> resources presets -> user overrides
> 
> ```
> chartValues := defaultChartValues(clusterName, apiKey.ApiKey)
> if sentryHelmContext.ResourcesPresets, err = helm.TuneResourcesValues(&chartValues, adequateNodesReports); err != nil {
> 	return err
> }
> if sentryHelmContext.ValuesOverride, err = helm.LoadChartValuesOverrides(&chartValues, viper.GetStringSlice(VALUES_FLAG)); err != nil {
> 	return err
> }
> ```

ok, then those functions needs to have more meaningful  names in term of signaling that they are settings values, and the sentry context should be set afterwards , to make it clear that are to steps - 1) overridevalues 2) set sentry context and not in the same row

its also preferable that TuneResourcesValues and LoadChartValuesOverrides will have the same prefix to signify they are injecting new values, I suggest something like SetResourceTunerOverrides  and SetUserOverrides

**Review by noamApps — APPROVED** [review]

**Pull request merged**